src/Aspire.Hosting.Azure.WebPubSub/AzureWebPubSubExtensions.cs

```diff
@@ -64,4 +93,78 @@ public static IResourceBuilder<AzureWebPubSubResource> AddAzureWebPubSub(this ID
                       .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
                       .WithManifestPublishingCallback(resource.WriteToManifest);
     }
+
+    /// <summary>
+    /// Add hub settings
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="hubName">The hub name.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<AzureWebPubSubHubResource> AddHub(this IResourceBuilder<AzureWebPubSubResource> builder, [ResourceName] string hubName)
+    {
+        var hubResource = new AzureWebPubSubHubResource(hubName, builder.Resource);
+        var hubBuilder = builder.ApplicationBuilder.CreateResourceBuilder(hubResource);
+        builder.Resource.Hubs[hubName] = hubResource;
+        return hubBuilder;
+    }
+
+    /// <summary>
+    /// Add event handler setting with expression
+    /// </summary>
+    /// <param name="builder">The builder for a Web PubSub hub.</param>
+    /// <param name="urlTemplateExpression">The expression to evaluate the URL template configured for the event handler.</param>
+    /// <param name="userEventPattern">The user event pattern for the event handler.</param>
+    /// <param name="systemEvents">The system events for the event handler.</param>
+    /// <param name="authSettings">The auth settings configured for the event handler.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<AzureWebPubSubHubResource> AddEventHandler(this IResourceBuilder<AzureWebPubSubHubResource> builder,
+        ReferenceExpression.ExpressionInterpolatedStringHandler urlTemplateExpression, string userEventPattern = "*", string[]? systemEvents = null, UpstreamAuthSettings? authSettings = null)
+    {
+        var urlExpression = ReferenceExpression.Create(urlTemplateExpression);
+
+        builder.Resource.EventHandlers.Add((construct, hub) =>
+        {
+            var hubName = hub.Name.Value;
+
+            var hubResource = hub.Properties.Value!;
+            if (urlExpression.ManifestExpressions.Count == 0)
+            {
+                var eventHandler = GetWebPubSubEventHandler(urlExpression.Format, userEventPattern, systemEvents, authSettings);
+
+                // when urlExpression is literal string, simply add
+                hubResource.EventHandlers.Add(eventHandler);
+            }
+            else
+            {
+                var count = hubResource.EventHandlers.Count;
```

Comment:
Is this right? Should the count be captured outside of this and read from the closure? These run after all of event handlers are added right? Won't they have the same value?

Reply:
let me rename hubResource, it is actually the CDK object..